app/api/api_v1/endpoints/search.py

```diff
@@ -1,39 +1,81 @@
 from typing import Any, List
 from fastapi import APIRouter, HTTPException
 from app import schemas
+from app.api import utils
+from app.api import index
 from app.api import deps
 
-from app.shared_state import existing_collections
+from app.core.config import settings  # pylint: disable=C0415
 
 router = APIRouter()
 
+import lxml.html
+import numpy as np
+
+
 @router.get("/", response_model = schemas.SearchResultFull)
 def search(query: str) -> Any:   
-    emptySearchResult =  schemas.SearchResult(id="", title="", description="", path="", locale="")
-    results = []
-    suggestions = []        # EMPTY FOR NOW
+    # A dictionary to store top_k results for each question across all collections
+    top_results = {}
 
+    question_results = []
+    results = []
+    answers = []
+    links = []
+    # Generate questions from the query
+    questions = utils.generate_questions(query, n = utils.N_QUESTIONS)
+    existing_collections = [x.decode('utf-8') for x in deps.db.connection.execute_command('FT._LIST')]
+    #print("Find Existing Collections!")
     if len(existing_collections) < 1:
         raise HTTPException(status_code=404, detail="No Collections")
 
-    for collection_name, collection in existing_collections.items():
-        if len(collection.documents) < 1:
-            continue
-        for document_id, document in collection.documents.items():
-            if query.lower() == document.title.lower():
-                result = schemas.SearchResult(
-                    id=f"{collection_name}_{document_id}",  # ID IS PATH FOR NOW
-                    title=document.title,
-                    description=document.description,
-                    path=document_id,
-                    locale=document.localeCode
-                )
-                results.append(result)
-
-    return schemas.SearchResultFull(
-        results=results,
-        suggestions=suggestions,
-        totalHits=len(results) + len(suggestions)
-    )
+    for question in questions:
+        query_embedding = utils.encode_blocks(deps.db.embedding, question)
+        # A list to store the search results for the current question from all collections
+        for collection in existing_collections:
+            #print(f"Search in {collection} ")
+            redis_search_results = index.search_vectors_knn(redis_connect = deps.db.connection, 
```

Comment:
Redis_om
Consider to wrap `index.search_vectors_knn` with method that returns list of results Pydantic objects. I guess you can find in the source code of  [redis_om](https://github.com/redis/redis-om-python/blob/87d3a080adf6e9dc2bda1b38204b5b341876f16e/aredis_om/model/model.py#L1371)  examples on how to do it properly.